lib/src/acter_avatar.dart

```diff
@@ -162,245 +126,242 @@ class _ActerAvatar extends State<ActerAvatar> {
+        // render for Group DM
+        GroupDM: ((avatarInfo, groupAvatarsInfo, size, groupAvatarSize) {
+          return groupAvatarsInfo != null && groupAvatarsInfo.isNotEmpty
+              ? Stack(
+                  alignment: Alignment.bottomLeft,
+                  clipBehavior: Clip.none,
+                  children: [
+                    CircleAvatar(
+                      foregroundImage: avatar,
+                      onForegroundImageError: avatarError,
+                      radius: size ?? 24,
+                    ),
+                    Positioned(
+                      left: -7,
+                      bottom: -5,
+                      child: tooltipMsgWithChild(
+                        groupAvatarsInfo[0],
+                        CircleAvatar(
+                          foregroundImage: groupAvatarsInfo[0].avatar,
+                          radius: size ?? 24,
+                        ),
+                      ),
+                    ),
+                    groupAvatarsInfo.length > 1
+                        ? Positioned.fill(
+                            bottom: -5,
+                            child: Align(
+                              alignment: Alignment.bottomRight,
+                              child: Container(
+                                width: 15,
+                                height: 15,
+                                alignment: Alignment.center,
+                                decoration: const BoxDecoration(
+                                  color: Colors.green,
+                                  shape: BoxShape.circle,
+                                ),
+                                child: Text(
+                                  '+${groupAvatarsInfo.length - 1}',
+                                  style: const TextStyle(fontSize: 8),
+                                ),
                               ),
                             ),
-                          ),
-                        )
-                      : const SizedBox.shrink(),
-                ],
-              )
-            : CircleAvatar(
-                foregroundImage: avatar,
-                onForegroundImageError: avatarError,
-                radius: widget.size ?? 24,
-              );
-    }
+                          )
+                        : const SizedBox.shrink(),
+                  ],
+                )
+              : CircleAvatar(
+                  foregroundImage: avatar,
+                  onForegroundImageError: avatarError,
+                  radius: size ?? 24,
+                );
+        }));
   }
 
-  Widget renderSpaceParent(BuildContext context) {
-    if (widget.badgeSize == null) {
-      // nothing. ignore
-      return SizedBox.shrink();
-    }
-    final badgeSize = widget.badgeSize ?? 20;
-    if (widget.avatarsInfo == null || widget.avatarsInfo!.isEmpty) {
-      return SizedBox(
-        height: badgeSize,
-        width: badgeSize,
+  Widget renderBadges(
```

Comment:
Logically the code is pretty much same overall except for this part, here the code has been modified to support rendering multiple parent spaces .i.e. badges in plugin terms.